### ISSUE TITLE
sum operation over dimension dim doesn't work on cudatensor of dimension larger than 4 

### ISSUE THREAD

**Issue by pyadolla** [torch/cutorch#92]
use case:

m=torch.CudaTensor(2,3,4,5,6):fill(1)
reduce = torch.sum(m,dim)        --fails [torch.CudaTensor with no dimension]

where dim can be any integer in [1..5].

reduce=torch.sum(m)                 --/does/ work
